pkg/distributor/distributor.go

```diff
@@ -1848,7 +1850,7 @@ func (d *Distributor) ActiveSeries(ctx context.Context, matchers []*labels.Match
 
 		stream, err := client.ActiveSeries(ctx, req)
 		if err != nil {
-			if errors.Is(err, context.Canceled) {
+			if errors.Is(err, context.Canceled) || status.Code(err) == codes.Canceled {
```

Comment:
I have moved this code to `util` and reused it in this PR. Happy to have your suggestions in case you think we can improve naming and/or the location of this code (should it be a separate package?).